api/src/main/java/org/bahmni/module/events/listener/PatientAdvice.java

```diff
@@ -0,0 +1,38 @@
+package org.bahmni.module.events.listener;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.bahmni.module.events.model.Event;
+import org.openmrs.Patient;
+import org.openmrs.module.webservices.rest.web.ConversionUtil;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.springframework.aop.AfterReturningAdvice;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.ApplicationEventPublisherAware;
+import org.springframework.lang.NonNull;
+
+import java.lang.reflect.Method;
+
+public class PatientAdvice implements AfterReturningAdvice, ApplicationEventPublisherAware {
+	
+	private final Logger log = LogManager.getLogger(PatientAdvice.class);
+
+	private ApplicationEventPublisher eventPublisher;
+	
+	@Override
+	public void afterReturning(Object returnValue, Method method, Object[] arguments, Object target) {
+		
+		Patient patient = (Patient) returnValue;
+		
+		Object representation = ConversionUtil.convertToRepresentation(patient, Representation.FULL);
+		Event event = new Event("bahmni-patient", representation, patient.getUuid());
```

Comment:
I feel like the Event Name and Topic Names should be moved to a Constants file (so that in one place we can see all the Event types and Topic Names). What do you think? Shall we wait until we have more than one?

Reply:
Enums like this will work ?
PATIENT_CREATED_UPDATED("bahmni-patient");
    private final String topic;
public String topic() {
        return topic;
    }

Reply:
Yes. Ideally patient created and modified event types should be different I think.

Reply:
We tried but difficult to differentiate at this point of time.

Reply:
Why is that? Same API is called in code for register/update patient? Some more details please.

Reply:
Although REST APIs are different but they both call **PatientService.save** method underneath. We have created Advice on patientService.save method.